core/templates/domain/topic_viewer/read-only-topic-object.factory.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of ReadOnlyTopic from
+ * topic data domain objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { SkillSummary, SkillSummaryObjectFactory } from
+  'domain/skill/SkillSummaryObjectFactory';
+import { StoryReference, StoryReferenceObjectFactory } from
+  'domain/topic/StoryReferenceObjectFactory';
+import { StorySummary } from
+  'domain/story/StorySummaryObjectFactory';
+import { Subtopic, SubtopicObjectFactory } from
+  'domain/topic/SubtopicObjectFactory';
+
+export interface ISkillDescriptions {
+  [skillId: string]: string | null;
+}
+
+export interface IDegreesOfMastery {
+  [skillId: string]: number | null;
+}
+
+export class ReadOnlyTopic {
+  _topicName: string;
+  _topicId: string;
+  _canonicalStorySummaries: Array<StorySummary>;
+  _additionalStorySummaries: Array<StorySummary>;
+  _uncategorizedSkillSummaries: Array<SkillSummary>;
+  _subtopics: Array<Subtopic>;
+  _degreesOfMastery: IDegreesOfMastery;
+  _skillDescriptions: ISkillDescriptions;
+  _trainTabShouldBeDisplayed: boolean;
+
+  constructor(topicName: string, topicId: string,
```

Comment:
break line after parenthesis (The way you have written on line 115)

Reply:
yes